cmake/modules/west.cmake

```diff
@@ -95,3 +95,55 @@ else()
     set(WEST WEST-NOTFOUND CACHE INTERNAL "West")
   endif()
 endif()
+
+# Verify blobs fetched using west. If the sha256 checksum isn't valid, a fatal
+# error message is printed.
+# Paths are relative to the module's zephyr/blobs directory.
+#
+# Usage:
+#   west_blobs_verify([MODULE module]
+#                     [PATHS path [paths...]])
+#
+# Example:
+# west_blobs_verify()                   # verify all blobs
+# west_blobs_verify(MODULE my_module)   # verify all blobs in "my_module"
+# west_blobs_verify(PATHS img/file.bin) # verify a single path
+function(west_blobs_verify)
+  if("${WEST}" STREQUAL "WEST-NOTFOUND")
+    message(WARNING "${CMAKE_CURRENT_FUNCTION} called without west, skipping")
+    return()
+  endif()
+
+  cmake_parse_arguments(WEST_BLOBS_VERIFY "" "MODULE" "PATHS" ${ARGN})
+
+  execute_process(
+    COMMAND ${WEST} blobs list ${WEST_BLOBS_VERIFY_MODULE} -f "{status} {path}"
+    OUTPUT_VARIABLE BLOBS_LIST_OUTPUT
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+    COMMAND_ERROR_IS_FATAL ANY
+  )
+
+  string(REPLACE "\n" ";" BLOBS_LIST ${BLOBS_LIST_OUTPUT})
+
+  if(WEST_BLOBS_VERIFY_PATHS)
+    foreach(path ${WEST_BLOBS_VERIFY_PATHS})
+      message(STATUS "Verifying blob \"${path}\"")
```

Comment:
there are currently 91 bin blobs defined in upstream Zephyr modules.
Printing status for each just becomes noise, compared to more important information.

```suggestion
```

If you prefer to keep the message, then please downgrade it to `VERBOSE` instead of `STATUS`.